lisa/target.py

```diff
@@ -659,6 +703,66 @@ def install_tools(self, tools):
             self.target.install(tool)
             self._installed_tools.add(tool)
 
+    def build_kmodule(self, kernel_src, kmodule_path):
```

Comment:
I think I missed that one back when I first reviewed this PR, apologies for that.

I'm really not fond of having building logic within LISA. We used to have some bits of code that would try to build a kernel and install it on a target (while taking into account different setups like tftp and whatnot), and we agreed that this was outside of the scope of LISA.

ATM, to get the load tracking tests going, you need to have the proper "proprietary" TP patches cherry-picked to your kernel, and have that kernel running on the target. I'm fine with having logic to load the modules, but not to build them - that should be left up to the user (just like building the kernel is left to him/her).

This sort of matches the logic we have for tools ATM. You can specify that you want to load `trace-cmd`, but we don't build it from source every time we do that. So here I would suggest taking as input a *compiled* module path to load it, but leave the building outside.

It also makes sense for CI flows, because it builds the whole thing anyway (kernel + modules), so it would just be plumbing of artifacts we already generate. This would avoid duplicating code that takes care of the building.

Reply:
I can appreciate that point of view but I think it's either Lisa do full management of the module or the user. Splitting it here and there is a bad idea and will open a worse can of worms.

As a middle ground I'd be happy to provide a patch instead + a script to apply it into a tree then tell the user to rebuild their kernel with this change before running Lisa. The module is tiny and we can enforce it to always be builtin and that will be that least effort solution if the user has to build the module anyway. There's little or no gray areas here and since the code is completely isolated it won't conflict with anything and should be guaranteed to apply cleanly except to small conflicts in Makefile maybe.

The advantage of current solution is that it is generic to serve other purposes and Douglas showed some interest in that already, but I don't mind as we choose something that is consistent. ie: either user has full responsibility or Lisa, but not divide it.